flake8_trio/__init__.py

```diff
@@ -53,6 +74,44 @@ def cst_parse_module_native(source: str) -> cst.Module:
     return mod
 
 
+def main():
+    parser = ArgumentParser(prog="flake8_trio")
+    parser.add_argument(
+        nargs="*",
+        metavar="file",
+        dest="files",
+        help="Files(s) to format, instead of autodetection.",
+    )
+    Plugin.add_options(parser)
+    args = parser.parse_args()
+    Plugin.parse_options(args)
+    if args.files:
+        # TODO: go through subdirectories if directory/ies specified
+        all_filenames = args.files
+    else:
+        # Get all tracked files from `git ls-files`
+        try:
+            root = os.path.relpath(_get_git_repo_root())
+            all_filenames = subprocess.run(
+                ["git", "ls-files"],
+                check=True,
+                timeout=10,
+                stdout=subprocess.PIPE,
+                text=True,
+                cwd=root,
+            ).stdout.splitlines()
+        except (subprocess.SubprocessError, FileNotFoundError):
+            print("Doesn't seem to be a git repo; pass filenames to format.")
+            sys.exit(1)
+        all_filenames = [
+            os.path.join(root, f) for f in all_filenames if _should_format(f)
+        ]
+    for file in all_filenames:
+        plugin = Plugin.from_filename(file)
+        for error in sorted(plugin.run()):
+            print(f"{file}:{error}")
```

Comment:
A small `ProcessPoolExecutor` invocation here can make it n_cores times faster for large repos... although we'd still want to sort by file and lineno.  Let's do that later.